src/CoreApi/BootstrapApi.cs

```diff
@@ -1,63 +1,74 @@
             var a = addrs.FirstOrDefault();
-            if (a == null)
+            if (a is null)
                 return null;
-            return new MultiAddress((string)a);
+            return new MultiAddress((string)a!);
         }
 
-        public async Task<IEnumerable<MultiAddress>> AddDefaultsAsync(CancellationToken cancel = default(CancellationToken))
+        public async Task<IEnumerable<MultiAddress>> AddDefaultsAsync(CancellationToken cancel = default)
         {
             var json = await ipfs.DoCommandAsync("bootstrap/add/default", cancel);
-            var addrs = (JArray)(JObject.Parse(json)["Peers"]);
+            var addrs = (JArray?)(JObject.Parse(json)["Peers"]);
+            if (addrs is null)
+            {
+                return Enumerable.Empty<MultiAddress>();
+            }
+
             return addrs
-                .Select(a => MultiAddress.TryCreate((string)a))
-                .Where(ma => ma != null);
+                .Select(a => MultiAddress.TryCreate((string)a!))
+                .Where(ma => ma is not null)
+                .Cast<MultiAddress>();
         }
 
-        public async Task<IEnumerable<MultiAddress>> ListAsync(CancellationToken cancel = default(CancellationToken))
+        public async Task<IEnumerable<MultiAddress>> ListAsync(CancellationToken cancel = default)
         {
             var json = await ipfs.DoCommandAsync("bootstrap/list", cancel);
-            var addrs = (JArray)(JObject.Parse(json)["Peers"]);
+            var addrs = (JArray?)(JObject.Parse(json)["Peers"]);
+            if (addrs is null)
+            {
+                return Enumerable.Empty<MultiAddress>();
+            }
+
             return addrs
-                .Select(a => MultiAddress.TryCreate((string)a))
-                .Where(ma => ma != null);
+                .Select(a => MultiAddress.TryCreate((string)a!))
+                .Where(ma => ma is not null)
+                .Cast<MultiAddress>();
         }
 
-        public Task RemoveAllAsync(CancellationToken cancel = default(CancellationToken))
+        public Task RemoveAllAsync(CancellationToken cancel = default)
         {
             return ipfs.DoCommandAsync("bootstrap/rm/all", cancel);
         }
 
-        public async Task<MultiAddress> RemoveAsync(MultiAddress address, CancellationToken cancel = default(CancellationToken))
+        public async Task<MultiAddress?> RemoveAsync(MultiAddress address, CancellationToken cancel = default)
         {
             var json = await ipfs.DoCommandAsync("bootstrap/rm", cancel, address.ToString());
-            var addrs = (JArray)(JObject.Parse(json)["Peers"]);
+            var addrs = (JArray?)(JObject.Parse(json)["Peers"]);
             var a = addrs.FirstOrDefault();
-            if (a == null)
+            if (a is null)
                 return null;
-            return new MultiAddress((string)a);
+            return new MultiAddress((string)a!);

```

Comment:
Don't need this null suppression, the `return` just above should handle it for us.
```suggestion
            return new MultiAddress((string)a);

```